### ISSUE TITLE
pre/post data processors to range/domain?

### ISSUE THREAD

**Issue by rijobro** [UCL/STIR#610]
Might be worth considering changing our pre/post data processors to using range and domain terminology.

That way, pre-forward and post-back data processors would both have the same name. They could then be set via the projector pair, and still be explicit as to what they'll act on.

**Comment by KrisThielemans** [comment]
I certainly agree to start using the range and domain terminology.

I'm confused about your comment on pre-forward and post-back data processors. They already have the same name. In fact, in both cases, the range and domain are going to be vector-space of the image.

**Comment by rijobro** [comment]
> we don't have pre/post data processors. We just have DataProcessor

Yes we only have `DataProcessor`, but we can choose to use them in image or sinogram space, before or after the projection. This leads to 4 possible combinations, right?

1. pre-forward projection (image)
2. post-back projection (image)
3. post-forward projection (sinogram)
4. pre-back projection (sinogram, hasn't been implemented)

What I'm saying is that in terms of methods, we can group the data processors that act in image (domain) and sinogram (range) space together.

If I want to use motion, I would do:

```
forward_projector->set_pre_data_processor(motion)
back_projector->set_post_data_processor(motion->adjoint)
```

whereas it could be:

```
forward_projector->set_domain_data_processor(motion)
back_projector->set_domain_data_processor(motion->adjoint)
```

And then the projector pair could be simply:
```
ProjectorPair::set_domain_data_processor()
```
and
```
ProjectorPair::set_range_data_processor()
```

As we've previously discussed, when using the projector pair method, the linear data processors would use the adjoint in the back projection.
